Documentation/faq.md

```diff
@@ -8,10 +8,50 @@
 
 `advertise-urls` specifies the addresses etcd clients or other etcd members should use to contact the etcd server. The advertise addresses must be reachable from the remote machines. Do not advertise addresses like `localhost` or `0.0.0.0` for a production setup since these addresses are unreachable from remote machines.
 
+
 ### Operation
 
+#### Why odd number of cluster size?
+
+A quorum is the majority of nodes in a cluster. When n is the cluster size, quorum is (n/2)+1. 2 is the quorum both for 3 and 4. Since Raft cluster becomes unavailable if the quorum of nodes is unavailable, adding another node to 3-node cluster does not add any value in terms of fault tolerance. And adding 1 node to single-node cluster increases the quorum size and causes leader election; if a wrong node were added by mistake, the cluster would lose its leader and not be able to make any progress. Always start with an odd number of nodes.
+
+#### What is maximum cluster size?
+
+Theoretically, there is no hard limit. However, we recommend no more than 7 nodes. [Google Chubby lock service][chubby], which is predecessor of etcd, typically runs 5 nodes in each cell. A 7-member cluster can provide enough fault tolerance in most cases. While larger cluster provides better fault tolerance the write performance reduces since data needs to be replicated to more machines.
```

Comment:
chubby, which is a similar project to etcd and widely deployed within Google for many years, suggests running 5 nodes. A 5-member etcd cluster can provide ...